test/integration_test.dart

```diff
@@ -915,15 +915,12 @@ Future<void> main() async {
 
           // Scroll to previous week
           await tester.drag(find.text('Tue'), Offset(size.width - 30, 0));
-          await tester.pumpAndSettle();
 
           // Expect previous week
-          final previousWeek = WeekYearRules.iso
-              .getWeekOfWeekYear(LocalDate.today().subtractWeeks(1));
-          expect(
-              find.byWidgetPredicate((widget) =>
-                  widget is WeekIndicator &&
-                  widget.week.toString() == previousWeek.toString()),
+
+          await tester.pumpAndSettle();
+
+          expect(find.byWidgetPredicate((widget) => widget is LeadHeader),
```

Comment:
This is a very bad change (same for the ones below). It essentially deletes a test that was checking that the weeks are displayed correctly, and instead tests... nothing. This `expect` call literally does nothing.

Remember: fixing tests that fail does NOT mean deleting the tests that fail :)

Instead, this should specifically test that the LeadHeader works correctly - first, set the setting to show the year week number like it did before (and keep the same logic, with checking the week number using getWeekOfWeekYear). Then copy the same test, change the setting to show academic week and test that it correctly shows the weeks and the holidays.

Now, to actually get the week number from the new widget it's a bit more tricky. I'm not sure what the best approach would be here (maybe the docs have some info about finding text in a specific widget?). I do have one idea, if we want to do something similar to what we had previously (where we're essentially finding the WeekIndicator widget and directly checking its `week` attribute; this can be done because WeekIndicator is stateless, but in our case LeadHeader is not and we can't directly access this attribute). We can create our own stateless widget (we can even call it WeekIndicator as well) from the Text widget in LeadHeader - basically in LeadHeader have something like:

```
child: 
     WeekIndicator(
        week: calendar == null ? '' : calendar.getWeekNumber(widget.date),
      ),
```

And then WeekIndicator would be a simple StatelessWidget, for which we can directly check the .week attribute. This way, when we use `find` to get the corresponding widget, we can test that it has the right value.

Again this is just an idea, there may be a better way to do this.